interop-cats/jvm/src/main/scala/scalaz/zio/interop/stm/STM.scala

```diff
@@ -0,0 +1,223 @@
+import scalaz.zio.ZIO
+import scalaz.zio.stm.{ STM => ZSTM }
+
+import scala.util.Try
+
+/**
+ * See [[scalaz.zio.stm.STM]]
+ */
+final class STM[F[+ _], +A] private[stm] (private[stm] val underlying: ZSTM[Throwable, A])(
+  implicit liftIO: ZIO[Any, Throwable, ?] ~> F
+) {
+  self =>
+
+  /**
+   * See `<*>` [[scalaz.zio.stm.STM]] `<*>`
+   */
+  final def <*>[B](that: => STM[F, B]): STM[F, (A, B)] =
+    self zip that
+
+  /**
+   * See [[scalaz.zio.stm.STM]] `<*`
+   */
+  final def <*[B](that: => STM[F, B]): STM[F, A] =
+    self zipLeft that
+
+  /**
+   * See [[scalaz.zio.stm.STM]] `*>`
+   */
+  final def *>[B](that: => STM[F, B]): STM[F, B] =
+    self zipRight that
+
+  /**
+   * See [[scalaz.zio.stm.STM]] `>>=`
+   */
+  final def >>=[B](f: A => STM[F, B]): STM[F, B] =
+    self flatMap f
+
+  /**
+   * See [[scalaz.zio.stm.STM#collect]]
+   */
+  final def collect[B](pf: PartialFunction[A, B]): STM[F, B] = new STM(underlying.collect(pf))
+
+  /**
+   * See [[scalaz.zio.stm.STM#commit]]
+   */
+  final def commit: F[A] = liftIO(underlying.commit)
+
+  /**
+   * See [[scalaz.zio.stm.STM#const]]
+   */
+  final def const[B](b: => B): STM[F, B] = self map (_ => b)
+
+  /**
+   * See [[scalaz.zio.stm.STM#either]]
+   */
+  final def either: STM[F, Either[Throwable, A]] = new STM(underlying.either)
+
+  /**
+   * See [[scalaz.zio.stm.STM#filter]]
+   */
+  final def filter(f: A => Boolean): STM[F, A] =
+    collect {
+      case a if f(a) => a
+    }
+
+  /**
+   * See [[scalaz.zio.stm.STM#flatMap]]
+   */
+  final def flatMap[B](f: A => STM[F, B]): STM[F, B] = new STM(underlying.flatMap(f.andThen(_.underlying)))
+
+  /**
+   * See [[scalaz.zio.stm.STM#flatten]]
+   */
+  final def flatten[B](implicit ev: A <:< STM[F, B]): STM[F, B] =
+    self flatMap ev
+
+  /**
+   * See [[scalaz.zio.stm.STM#fold]]
+   */
+  final def fold[B](f: Throwable => B, g: A => B): STM[F, B] = new STM(underlying.fold(f, g))
+
+  /**
+   * See [[scalaz.zio.stm.STM#foldM]]
+   */
+  final def foldM[B](f: Throwable => STM[F, B], g: A => STM[F, B]): STM[F, B] =
+    new STM(underlying.foldM(f.andThen(_.underlying), g.andThen(_.underlying)))
+
+  /**
+   * See [[scalaz.zio.stm.STM#map]]
+   */
+  final def map[B](f: A => B): STM[F, B] = new STM(underlying.map(f))
+
+  /**
+   * See [[scalaz.zio.stm.STM#mapError]]
+   */
+  final def mapError[E1 <: Throwable](f: Throwable => E1): STM[F, A] = new STM(underlying.mapError(f))
+
+  /**
+   * See [[scalaz.zio.stm.STM#option]]
+   */
+  final def option: STM[F, Option[A]] =
+    fold[Option[A]](_ => None, Some(_))
+
+  /**
+   * See [[scalaz.zio.stm.STM#orElse]]
+   */
+  final def orElse[A1 >: A](that: => STM[F, A1]): STM[F, A1] = new STM(underlying.orElse(that.underlying))
+
+  /**
+   * See [[scalaz.zio.stm.STM#orElseEither]]
+   */
+  final def orElseEither[B](that: => STM[F, B]): STM[F, Either[A, B]] =
+    (self map (Left[A, B](_))) orElse (that map (Right[A, B](_)))
+
+  /**
+   * See scalaz.zio.stm.STM.unit
```

Comment:
No Scaladoc link?

Reply:
I did this because in this case `sbt doc` command fails with err message like `Could not find any member to link for "scalaz.zio.stm.STM.unit".`   I suspect this is because it sees `unit` in the `STM` class and also in companion object, and fails to resolve which `unit` to link.

Reply:
No worries!